### DISCUSSION TITLE
mdx file is not shown with the rest of the stories

### DISCUSSION THREAD

**Discussion by carloscervera120571** [storybookjs/storybook#25074] [Help]
### Summary

I have an mdx file that I can't get to display along with the rest of the stories in ts format. You can see it in the attached image. How can I get it?
![recorte menu storybook](https://github.com/storybookjs/storybook/assets/132054846/5bbdca7e-2582-44e1-8721-6f61d03d1146)

### Additional information

Version: 7.5.3

Main.ts:

import type { StorybookConfig } from "@storybook/angular";

const config: StorybookConfig = {
  stories: ["../src/**/*.mdx", "../src/**/*.stories.@(js|jsx|mjs|ts|tsx)", "../projects/**/*.stories.@(js|jsx|mjs|ts|tsx)"],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-interactions"    
  ],
  framework: {
    name: "@storybook/angular",
    options: {},
  },
  docs: {
    autodocs: "tag",
  },
  staticDirs: ['../src/stories/assets'],
  previewHead: (head) => `
  ${head}
  ${
    '<script src="https://www.ccoo.es/js/multi.js"></script>'
  }
`,
};
export default config;

**Comment by Marklb** [comment]
Without seeing your project or a reproduction, it is hard to say for sure. If you are not setting a `title` in `Meta` for the files then maybe try moving the MDX or setting `title` in `Meta`. The docs explain how the auto-titles work here: https://storybook.js.org/docs/configure/sidebar-and-urls#csf-30-auto-titles

If you are not using auto-titles then you should just have to adjust the title of your MDX to not start with "stories/".

**Reply by carloscervera120571** [reply]
The problem is that my mdx cannot have a meta tag, because it does not contain a story. It will have text only. For now I have tried the one that comes as an example:

#GettingStarted

Welcome! Whether you're a designer or a developer, this guide will help you get started and connect you to the essential resources you need.

## Table of Contents

- [Design Resources](#design-resources)
  - [Figma](#figma)
  - [UI/UX Design Guidelines](#uiux-design-guidelines)
  - [Design Assets](#design-assets)

- [Development Resources](#development-resources)
  - [Coding Standards](#coding-standards)
  - [Version Control](#version-control)
  - [Development Tools](#development-tools)

---

## Design Resources

### Figma

[Figma](https://www.figma.com/) is a collaborative design and prototyping tool. It's the heart of the design process, allowing designers to work together seamlessly.

- **Get Access**: If you're not already part of the Figma project, request access from the project lead or manager.

### UI/UX Design Guidelines

Before you dive into designing, familiarize yourself with our UI/UX design guidelines. They provide valuable insights into our design philosophy and standards.

- [UI/UX Guidelines Document](https://your-design-guidelines-link.com)

### Design Assets

All the essential design assets like logos, icons, and brand guidelines can be found in the Figma project. Ensure you have access and familiarize yourself with these assets for consistency.

---

## Development Resources

### Coding Standards

Maintaining a consistent code style is essential for collaborative development. Our coding standards document will guide you on best practices.

- [Coding Standards Document](https://your-coding-standards-link.com)

### Version Control

We use Git for version control. Make sure you have Git installed and are familiar with its basics.

### Development Tools

Your development environment is critical. Here are some tools and resources to help you set up your workspace:

- **Code Editor**: We recommend using [Visual Studio Code](https://code.visualstudio.com/) for development. It's highly customizable and supports a wide range of extensions.

- **Package Manager**: [npm](https://www.npmjs.com/) is the package manager we use for JavaScript projects. Install it to manage project dependencies.

**Reply by kylegach** [reply]
@carloscervera120571, can you share a reproduction?

From the screenshot and your comments, I'm guessing your filesystem looks something like:

```
- src
  - Alerts
    - Alerts.stories.jsx
  - Badges
    - Badges.stories.jsx
  ...
  - stories
    - test.mdx
```

As Marklb suggested, if you move `test.mdx` directly under the `src` directory, it should display where you expect. Or, you can add `<Meta title="Test" />`.

-----

> The problem is that my mdx cannot have a meta tag, because it does not contain a story.

That's incorrect. You're right that MDX docs without stories cannot have a `<Meta of={...} />` block. But it can have a `<Meta title="..." />` block.

https://storybook.js.org/docs/writing-docs/mdx#writing-unattached-documentation

https://storybook.js.org/docs/api/doc-block-meta#attached-vs-unattached

**Comment by carloscervera120571** [comment]
Thank you so much. It's true about the meta title, but it still comes out disordered, now it shows it at the top. On the other hand, all the .stories and .mdx files are in the stories directory

**Reply by kylegach** [reply]
> still comes out disordered, now it shows it at the top

That's because of your `stories` glob array:

```js
stories: ["../src//*.mdx", "../src//.stories.@(js|jsx|mjs|ts|tsx)", "../projects/**/.stories.@(js|jsx|mjs|ts|tsx)"],
```

Because `*.mdx` is only in the first item of that array, it will always be sorted first.

But, I don't think that's an issue because:

> On the other hand, all the .stories and .mdx files are in the stories directory

Aha. So you're not using autotitles, then. Instead, you're defining a `title` in each stories file.

Can you share an example stories file title and the one you're trying to use for the MDX file?

**Comment by carloscervera120571** [comment]
Adjunto contenido actual del mdx:

import { Meta } from '@storybook/blocks';

<Meta title="Introducción" />

# Definition

Button is a clickable interactive element that triggers a response.

You can place text and icons inside of a button.

Buttons are often used for form submissions and to toggle elements into view.

## Usage

The component comes in different variants such as `primary`, `secondary`, `large` and `small` which you can use to alter the look and feel of the button.

## Inputs

Button has the following properties:

Y el resultado al ejecutar:

![Captura](https://github.com/storybookjs/storybook/assets/132054846/0105bb76-c597-46ad-aa04-f20f44b0c624)

**Reply by kylegach** [reply]
Your code will display as code if you surround it in 3 backticks to start and end, like this:

````mdx
```mdx
import { Meta } from '@storybook/blocks';

# Definition

Button is a clickable interactive element that triggers a response.
```
````

**Comment by carloscervera120571** [comment]
Thank you very much for the help

Example of a story title:

```storie
const meta: Meta<InputTextComponent> = {
    title: 'Input Text',

Example of a mdx title

```mdx
<Meta title="Introducción" />
